.github/workflows/pypi-publish.yaml

```diff
@@ -16,22 +16,21 @@ jobs:
     steps:
       - uses: actions/checkout@v4
 
-      - name: Install Poetry
-        run: |
-          pipx install poetry
-          pipx inject poetry "poetry-dynamic-versioning[plugin]"
+      - name: Install uv
+        uses: astral-sh/setup-uv@v5
```

Comment:
Would you be able to use pipx here?

Reply:
Yes. Installing uv via pipx works as well. Astral suggest to use the action, https://docs.astral.sh/uv/guides/integration/github/

Reply:
I understand that uv can also be a replacement for pipx, but I could use some hand holding.  I also see pipx being used to install uv here: https://docs.astral.sh/uv/getting-started/installation/#standalone-installer and since its become our defacto installer, I'd like to understand why we don't use it and/or what the migration path looks like to replace it with uv.  This should not block this PR.

Reply:
Should we have a standard bit of blurb that we add to all the migrating projects -- something like

> This project uses [uv](https://docs.astral.sh/uv/) for dependency management. See the [uv documentation](https://docs.astral.sh/uv/) for uv installation and usage instructions.

Reply:
For developers on their local machines, I would advocate for using uv's standalone installer over the `pipx` approach because it breaks the chain of tool dependencies. That is, it eliminates insanity like "to install uv you need pipx, to install pipx you need Python, to install Python you need pyenv". Stop the madness! Just use the uv standalone installer, now you have uv. It can install Python for you. It can also act as a `pipx` replacement for you.

For GitHub actions, if their docs recommend the `astral-sh/setup-uv`, then we should use it.